### PR TITLE
cmake: Grab TORCH_DEFAULT_VERSION from version.txt

### LINKED ISSUE #35210: libTorch version not matching checked out version when building with build_libtorch.py
I have noticed that when building libtorch using build_libtorch script on the newest version, the TorchVersion file indicates it builds Torch 1.1. However if i build pytorch using the setup.py build command, the libtorch version will match the checked out version. Whats going on here?

Also where are the required libTorch libs needed for standalone c++ when you do setup.py build? the documentation indicates ../../tmp_install but i cannot even find that path

This happens on any linux setup i have tried.

cc @ezyang @seemethere

### PR DESCRIPTION
**Pull request by seemethere**
This variable hasn't been updated in a long time since it usually just
gets overwritten by whatever is in the setup.py but let's set the
default to something a bit more in-line with what we're actually
building.

Closes https://github.com/pytorch/pytorch/issues/35210

cc @ksasso1028

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

### CODE REVIEW

**Review by malfet — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -363,7 +363,7 @@ include(cmake/Utils.cmake)
 include(cmake/public/utils.cmake)
 
 # ---[ Version numbers for generated libraries
-set(TORCH_DEFAULT_VERSION "1.1.0")
+file(READ version.txt TORCH_DEFAULT_VERSION)
```
Comment (malfet) [inline CMakeLists.txt]: Can you please set it to some default value, if read failed for some reason?
```suggestion
file(READ version.txt TORCH_DEFAULT_VERSION)
if ("${TORCH_DEVALUE_VERSION} " STREQUAL " ")
  message(WARNING "Could not get version from 'version.txt'")
  set(TORCH_DEFAULT_VERSION "1.5.0")
endif()
```

**Comment by ksasso1028** [comment]
awesome  work

File: CMakeLists.txt (hunk 1 above)
Comment (seemethere) [reply CMakeLists.txt:366]: Should be resolved by a follow up push :+1:

**Comment by JackCaoG** [comment]
xla test failure is due to kl_div in this branch is out of date but it is pulling the latest xla.
```
Apr 01 23:38:02 AtenXlaType function missed override: Tensor kl_div(const Tensor& self, const Tensor& target, int64_t reduction, bool log_target); // kl_div(Tensor,Tensor,int64_t,bool)->Tensor
Apr 01 23:38:02 AtenXlaType function missed override: Tensor kl_div_backward(const Tensor& grad_output, const Tensor& self, const Tensor& target, int64_t reduction, bool log_target); // kl_div_backward(Tensor,Tensor,Tensor,int64_t,bool)->Tensor
```

should be safe to ignore.

**Pull request closed without merging**